apiserver/internal/handlers/resources/resourcemigration.go

```diff
@@ -91,120 +83,84 @@ func (h *resourcesMigrationUploadHandler) servePost(w http.ResponseWriter, r *ht
 func (h *resourcesMigrationUploadHandler) processPost(
 	r *http.Request,
 	resourceService ResourceService,
-	applicationService ApplicationService,
 ) (coreresource.Resource, error) {
 	var empty coreresource.Resource
 	ctx := r.Context()
 	query := r.URL.Query()
+	resourceName := query.Get("name")
+	if resourceName == "" {
+		return empty, errors.BadRequestf("missing resource name")
+	}
+	if isPlaceholder(query) {
+		// If the resource is a placeholder, do nothing. Information about
+		// resources without an associated blob is also migrated during the
+		// database migration, there is nothing to do here.
+		return empty, nil
+	}
+	appName := query.Get("application")
+	if appName == "" {
+		// If the application name is empty, do nothing. Information about unit
+		// resources is also migrated during the database migration. There is
+		// nothing to do here.
+		return empty, nil
+	}
 
-	userID := query.Get("user") // Is allowed to be blank.
+	// Get the resource UUID.
+	resUUID, err := resourceService.GetResourceUUIDByApplicationAndResourceName(
+		ctx, appName, resourceName,
+	)
+	if err != nil {
+		return empty, internalerrors.Errorf(
+			"resource upload failed: getting resource %s on application %s: %w",
+			appName, resourceName, err,
+		)
+	}
 
-	// Get the target of the upload, which is an application with or without
-	// unit.
-	target, err := getUploadTarget(ctx, applicationService, query)
+	retrievedBy, retrievedByType := determineRetrievedBy(query)
+	err = resourceService.StoreResource(ctx, resource.StoreResourceArgs{
+		ResourceUUID:    resUUID,
+		Reader:          r.Body,
```

Comment:
question: Where do we verify that the fingerprint and size sent, match the actual blob received?